### PR TITLE
feat: Phase 3.1 - Static Routes Generator

### PR DESCRIPTION
**Pull request by duplico**
## Summary

Implements Phase 3.1 of the vyos-onecontext implementation plan: Static Routes Generator.

This PR adds support for parsing and generating VyOS configuration commands from `ROUTES_JSON` context variable, enabling operators to define static routes for VyOS routers via OpenNebula contextualization.

## Changes

**Generator Implementation:**
- Added `StaticRoutesGenerator` class in `src/vyos_onecontext/generators/routing.py`
- Integrated into `generate_config()` pipeline in `generators/__init__.py`
- Parses `RoutesConfig` model from `ROUTES_JSON`
- Generates VyOS commands following Sagitta syntax

**Route Types Supported:**
1. **Gateway routes**: Routes with next-hop IP address
   - Example: `set protocols static route 10.96.0.0/13 next-hop 10.63.255.1`
2. **Interface routes**: Routes directly via an interface (no next-hop)
   - Example: `set protocols static route 192.168.0.0/16 interface eth2`
3. **VRF routes**: Routes assigned to a specific VRF
   - Example: `set vrf name management protocols static route X/Y next-hop Z`
4. **Custom administrative distance**: Non-default distance values
   - Example: `... distance 10`

**Testing:**
- 15 comprehensive unit tests covering all route types
- Tests for edge cases (no routes, empty list, default distance handling)
- Tests for VRF assignment and mixed configurations
- All tests pass (325 total tests, including existing)

## Validation

```bash
just check
```

All checks pass:
- Linting (ruff): No issues
- Type checking (mypy): No issues
- Unit tests (pytest): 325 tests passed

## Design Decisions

1. **Default distance handling**: Administrative distance defaults to 1 (VyOS default). When distance=1, we don't emit the distance command to keep configuration clean.
2. **VRF support**: Routes can be assigned to VRFs using the `vrf` field. This is essential for management VRF routing.
3. **Gateway vs Interface routes**: Routes can specify either `gateway` (next-hop) OR use interface routing (no gateway). Both patterns are valid in VyOS Sagitta.

## Context Reference

The `ROUTES_JSON` schema is documented in `docs/context-reference.md` (lines 152-202).

Example Terraform usage:
```hcl
ROUTES_JSON = jsonencode({
  static = [
    { interface = "eth1", destination = "0.0.0.0/0", gateway = "10.63.255.1" },
    { interface = "eth2", destination = "10.96.0.0/13", gateway = "10.69.100.1" },
    { interface = "eth0", destination = "192.168.0.0/16", gateway = "10.0.1.254", vrf = "management" }
  ]
})
```

## Test Plan

- [x] Unit tests for all route types pass
- [x] Linting passes (ruff)
- [x] Type checking passes (mypy)
- [x] Integration with existing generators works correctly
- [ ] Manual testing with real VyOS router (integration test)
- [ ] End-to-end test with OpenNebula context

## Related

- Closes issue for Phase 3.1 implementation
- Part of larger vyos-onecontext v3 project
- See `docs/design.md` and implementation plan for context

Generated with Claude Code (Sonnet 4.5)

### CODE REVIEW

File: src/vyos_onecontext/generators/__init__.py [hunk 1]
```diff
@@ -41,6 +41,9 @@ def generate_config(config: RouterConfig) -> list[str]:
     # Routing (default gateway selection for non-management interfaces)
     commands.extend(RoutingGenerator(config.interfaces).generate())
 
+    # Static routes (ROUTES_JSON)
+    commands.extend(StaticRoutesGenerator(config.routes).generate())
+
     # VRF configuration (management VRF)
     commands.extend(VrfGenerator(config.interfaces).generate())
```
[the remark this answers is not in the document]
Comment (duplico) [reply src/vyos_onecontext/generators/__init__.py]: Fixed in commit 5456a58. Moved StaticRoutesGenerator after VrfGenerator to ensure VRF is created before routes can reference it. Updated generator order: interfaces -> routing -> VRF -> static routes -> SSH.

(AI-generated via Claude Code w/ Opus 4.5)

File: src/vyos_onecontext/generators/routing.py [hunk 2]
```diff
@@ -76,3 +77,65 @@ def _select_default_gateway(self) -> str | None:
...
+
+                # Add distance if non-default
+                if route.distance != 1:
+                    commands.append(f"{base} next-hop {route.gateway} distance {route.distance}")
+            else:
+                # Interface route (no next-hop)
+                commands.append(f"{base} interface {route.interface}")
+
+                # Add distance if non-default
+                if route.distance != 1:
```
[the remark this answers is not in the document]
Comment (duplico) [reply src/vyos_onecontext/generators/routing.py]: Fixed in commit 5456a58. Rewrote distance logic to generate only ONE command per route. When distance == 1 (default), generates base command without distance. When distance != 1, generates single command with distance parameter.

(AI-generated via Claude Code w/ Opus 4.5)

[the remark this answers is not in the document]
Comment (duplico) [reply src/vyos_onecontext/generators/routing.py]: Fixed in commit 5456a58. Same distance logic fix applied to interface routes. Only ONE command generated per route regardless of gateway or interface type.

(AI-generated via Claude Code w/ Opus 4.5)

File: tests/test_generators.py [hunk 3]
```diff
@@ -517,6 +516,319 @@ def test_generate_all_interfaces_management_vrf(self):
...
+                    distance=10,
+                )
+            ]
+        )
+        gen = StaticRoutesGenerator(routes)
+        commands = gen.generate()
+
+        assert len(commands) == 2
+        assert "set protocols static route 10.0.0.0/8 next-hop 10.63.255.1" in commands
+        assert "set protocols static route 10.0.0.0/8 next-hop 10.63.255.1 distance 10" in commands
```
[the remark this answers is not in the document]
Comment (duplico) [reply tests/test_generators.py]: Fixed in commit 5456a58. Updated test to expect len(commands) == 1 and verify single command with distance parameter.

(AI-generated via Claude Code w/ Opus 4.5)

File: tests/test_generators.py [hunk 4]
```diff
@@ -517,6 +516,319 @@ def test_generate_all_interfaces_management_vrf(self):
...
+                    distance=5,
+                )
+            ]
+        )
+        gen = StaticRoutesGenerator(routes)
+        commands = gen.generate()
+
+        assert len(commands) == 2
+        assert "set protocols static route 192.168.0.0/16 interface eth2" in commands
+        assert "set protocols static route 192.168.0.0/16 interface eth2 distance 5" in commands
```
[the remark this answers is not in the document]
Comment (duplico) [reply tests/test_generators.py]: Fixed in commit 5456a58. Updated test to expect len(commands) == 1 and verify single command with distance parameter.

(AI-generated via Claude Code w/ Opus 4.5)

File: tests/test_generators.py [hunk 5]
```diff
@@ -517,6 +516,319 @@ def test_generate_all_interfaces_management_vrf(self):
...
+
+        assert len(commands) == 2
+        assert (
+            "set vrf name management protocols static route 192.168.0.0/16 next-hop 10.0.1.254"
+            in commands
+        )
+        assert (
+            "set vrf name management protocols static route 192.168.0.0/16 "
+            "next-hop 10.0.1.254 distance 20" in commands
+        )
```
[the remark this answers is not in the document]
Comment (duplico) [reply tests/test_generators.py]: Fixed in commit 5456a58. Updated test to expect len(commands) == 1 and verify single command with distance parameter.

(AI-generated via Claude Code w/ Opus 4.5)

File: tests/test_generators.py [hunk 6]
```diff
@@ -517,6 +516,319 @@ def test_generate_all_interfaces_management_vrf(self):
...
+
+        assert len(commands) == 2
+        assert (
+            "set vrf name management protocols static route 192.168.0.0/16 interface eth0"
+            in commands
+        )
+        assert (
+            "set vrf name management protocols static route 192.168.0.0/16 "
+            "interface eth0 distance 5" in commands
+        )
```
[the remark this answers is not in the document]
Comment (duplico) [reply tests/test_generators.py]: Fixed in commit 5456a58. Updated test to expect len(commands) == 1 and verify single command with distance parameter.

(AI-generated via Claude Code w/ Opus 4.5)

File: tests/test_generators.py [hunk 7]
```diff
@@ -517,6 +516,319 @@ def test_generate_all_interfaces_management_vrf(self):
...
+
+
```
[the remark this answers is not in the document]
Comment (duplico) [reply tests/test_generators.py:823]: Fixed in commit 5456a58. Added test_generate_config_static_routes_with_management_vrf to TestGenerateConfig class. This integration test verifies VRF is created before static routes that reference it, and validates the correct command ordering.

(AI-generated via Claude Code w/ Opus 4.5)

File: tests/test_generators.py [hunk 8]
```diff
@@ -517,6 +517,271 @@ def test_generate_all_interfaces_management_vrf(self):
...
+                )
+            ]
+        )
+        gen = StaticRoutesGenerator(routes)
+        commands = gen.generate()
+
+        # Should only have the base next-hop command, not the distance command
+        assert len(commands) == 1
+        assert commands[0] == "set protocols static route 10.0.0.0/8 next-hop 10.63.255.1"
+
```
[the remark this answers is not in the document]
Comment (duplico) [reply tests/test_generators.py:775]: Fixed in commit 5456a58. Added test_generate_vrf_interface_route_no_gateway which tests VRF interface routes without a gateway (the missing test coverage you identified).

(AI-generated via Claude Code w/ Opus 4.5)
